tests/test_kernel_magics.py

```diff
@@ -417,3 +417,20 @@ def test_delete_with_force_different_session():
 
     spark_controller.get_session_id_for_client.assert_called_once_with(magic.session_name)
     spark_controller.delete_session_by_id.assert_called_once_with(magic.endpoint, session_id)
+
+
+@with_setup(_setup, _teardown)
+def test_add_session_throw_exception():
```

Comment:
A better name would be `test_start_session_displays_fatal_error_when_session_throws`